simulariumio/converter.py

```diff
@@ -279,6 +280,35 @@ def _determine_plot_reader(plot_type: str = "scatter") -> [Reader]:
 
         raise UnsupportedPlotTypeError(plot_type)
 
+    def _check_agent_ids_are_unique_per_frame(self) -> bool:
+        """
+        For each frame, check that none of the unique agent IDs overlap
+        """
+        bundleData = self._data["spatialData"]["bundleData"]
+        for t in range(len(bundleData)):
+            data = bundleData[t]["data"]
+            next_uid_index = 1
+            uids = []
+            get_n_subpoints = False
+            for i in range(len(data)):
+                if i == next_uid_index:
+                    # get the number of subpoints
+                    # in order to correctly increment next_uid_index
+                    if get_n_subpoints:
+                        next_uid_index += data[i] + 2
+                        get_n_subpoints = False
+                        continue
+                    # there should be a unique ID at this index, check for duplicate
+                    uid = data[i]
+                    if uid in uids:
+                        raise Exception(
+                            f"found duplicate ID {uid} in frame {t} at index {i}"
+                        )
+                    uids.append(uid)
+                    next_uid_index += 9
```

Comment:
If the # of sub-points were directly read, and the iterator assigned to the next uid (similar to line 298 but w/o the extra loop iteration), the structure of the code would better reflect the flow, making it easier to understand at a glance imo.